db/db_basic_test.cc

```diff
@@ -2101,6 +2097,47 @@ class DBBasicTestWithTimestampWithParam
   }
 };
 
+class DBBasicTestWithTimestamp
+    : public DBBasicTestWithTimestampBase {
+ public:
+  DBBasicTestWithTimestamp() {}
+
+ protected:
+  class TestComparator : public TestComparatorBase {
+   public:
+    explicit TestComparator(size_t ts_sz) : TestComparatorBase(ts_sz) {}
+
+    int CompareImpl(const Slice& a, const Slice& b) const override {
+      int n1 = atoi(a.data()+3);  // 3: length of "Key", to be skipped
```

Comment:
Is the value `3` specific to your test case? If so, can you use a public static const in TestComparator or somewhere else, to indicate the length of key?
`a` is a `Slice` object whose underlying `data()` does not have to be a `\0`-terminated, C-string.

Reply:
Made '3' a const. Good catch on the Slice data type. Fixed.
(I'm on vacation and response can be slow.)